File: Applications/AspCore/NLP/Models/AnalysisResult.cs
using Microsoft.AspNetCore.SignalR;
using System.Collections.Generic;

namespace NLP.Models
{
    public class AnalysisResult
    {
        public int AnalysisResultId { get; set; }

        public string Url { get; set; }
        public double AccuracyScore { get; set; }
        public string ConfusionMatrix { get; set; }

        public string UserId { get; set; }
        public User User { get; set; }

    }
}

File: Applications/react/nlp/src/models/AnalysisResult.ts
export interface IAnalysisResult {
    analysisResultId: number;
    url: string;
    accuracyScore: number
    confusionMatrix: string
    userId: number;
    user: object
}

export class AnalysisResult implements IAnalysisResult {
    analysisResultId: number;
    url: string;
    accuracyScore: number
    confusionMatrix: string
    userId: number;
    user: object

    constructor() {
        this.analysisResultId = 0;
        this.url = "";
        this.accuracyScore = 0;
        this.confusionMatrix = "";
        this.userId = 0;
        this.user = {};
    }
}